giftless/transfer/storage/google_cloud.py

```diff
@@ -0,0 +1,114 @@
+import os
+from datetime import datetime, timedelta, timezone
+from typing import Any, BinaryIO, Dict, Optional
+
+from google.cloud import storage  # type: ignore
+from google.cloud.exceptions import Conflict  # type: ignore
+
+from giftless.transfer.basic_external import ExternalStorage
+from giftless.transfer.basic_streaming import StreamingStorage
+from giftless.transfer.exc import ObjectNotFound
+
+
+class GoogleCloudBlobStorage(StreamingStorage, ExternalStorage):
+    """Google Cloud Storage backend supporting direct-to-cloud
+    transfers.
+
+    """
+
+    def __init__(self, bucket_name: str, api_key: Optional[str] = None,
+                 account_json_path: Optional[str] = None,
+                 path_prefix: Optional[str] = None, **_):
+        self.bucket_name = bucket_name
+        self.path_prefix = path_prefix
+        self.api_key = api_key
+        self.storage_client = storage.Client.from_service_account_json(
+            account_json_path)
+        self._init_container()
```

Comment:
Calling anything that may fail (throw exceptions etc.) in a constructor is considered a bad practice, and also makes testing harder.